src/Microsoft.ML.AutoML/ColumnInference/ColumnPurpose.cs

```diff
@@ -13,6 +13,7 @@ internal enum ColumnPurpose
         TextFeature = 4,
         Weight = 5,
         ImagePath = 6,
-        SamplingKey = 7
+        SamplingKey = 7,
+        LabelFeature,   // CategoricalFeature that requires ValueToKey converter, better naming?
```

Comment:
I think we'd want two new explicit column purposes of: 
* ColumnPurpose.UserId
* ColumnPurpose.ItemId

I don't know that we have a method of telling them apart otherwise. Feel free to correct me if wrong.